### ISSUE TITLE
OpenSSL's PEM->DER decoder should not pretend to know everything

### ISSUE THREAD

**Issue by levitte** [openssl/openssl#15929]
Our PEM->DER decoder assumes that a PEM starter that looks like this:

```
-----BEGIN SOMETHING PRIVATE KEY-----
```

... always denotes a *key type* "SOMETHING", and passes that on as information for further decoding, with the structure "type-specific"

However, the TPM2 projects are using the following PEM starter:

```
-----BEGIN TSS2 PRIVATE KEY-----
```

However, TSS2 isn't a key type, it's rather a wrapping structure indicator, just like `-----BEGIN PRIVATE KEY-----` is understood to contain a `PrivateKeyInfo` structure.

Bottom line, our PEM->DER decoder should not pretend to know all PEM names and their meaning.  This gives a chance to other PEM->DER decoders who do know the details better to parse things correctly.
